ci/README.md

```diff
@@ -0,0 +1,202 @@
+# Continuous Integration
+
+When a Pull Request (PR) is opened, several automated checks are performed to ensure the new compiler still builds, passes tests, and does not break external projects.
+Since DMD runs on many platforms and lots of D infrastructure depends on it, they are spread over many CI services.
+
+## Troubleshooting failed checks
+When your PR fails checks, click on the "Details" link next to it to see what the error is.
+
+Errors caused by the changes you made should be addressed by pushing new commits to your branch.
+
+When it looks like the error is unrelated to your PR's changes, try rebasing your branch.
+First, ensure that you have a 'remote' pointing to both dlang/dmd and your own fork of dmd, for example:
+```
+git remote -v
+origin  git@github.com:your-github-name/dmd.git (fetch)
+origin  git@github.com:your-github-name/dmd.git (push)
+upstream        git@github.com:dlang/dmd.git (fetch)
+upstream        git@github.com:dlang/dmd.git (push)
+```
+
+If you don't have an `upstream` remote, you can add it with:
+```
+git remote add upstream git@github.com:dlang/dmd.git
+```
+
+Then, pull the latest changes from the branch that your PR merged into (`master` or `stable`) and rebase your branch.
+```
+git checkout master
+git pull --ff-only upstream master
+git checkout your-branch
+git rebase master
+git push origin master --force
+```
+
+If your branch is already based on the latest commit, you get this message:
+```
+Current branch your-branch is up to date.
+```
+
+If you still want to re-run the checks, you can amend your last commit and force push:
+```
+git commit --amend
+git push origin master --force
+```
+
+When an unrelated failure persists, ask a maintainer for help.
+
+## Overview of checks
+
+There are currently 49 checks.
+
+### Azure pipelines
+
+**Config**: [azure-pipelines.yml](https://github.com/dlang/dmd/blob/master/azure-pipelines.yml)
+
+**Checks**:
+- Azure pipelines
+- Azure pipelines (Windows_Coverage x64)
+- Azure pipelines (Windows_DMD_bootstrap x64)
+- Azure pipelines (Windows_DMD_latest x64)
+- Azure pipelines (Windows_DMD_latest x86-OMF)
+- Azure pipelines (Windows_VisualD_LDC x64_Debug)
+- Azure pipelines (Windows_VisualD_LDC x86-mscoff)
+- Azure pipelines (Windows_VisualD_LDC x86-mscoff_MinGW)
+
+Azure pipelines run on Windows platforms, and build DMD, Phobos and Druntime, and run their unittests.
+Windows has three binary formats: 32-bit OMF (deprecated), 32-bit COFF, and 64-bit COFF.
+
+### DAutoTest
+
+**Config**: [CyberShadow/DAutoTest](https://github.com/CyberShadow/DAutoTest)
+
+**Checks**:
+- CyberShadow/DAutoTest — Documentation
+
+DMD includes a documentation generator, which is also used to build [the dlang.org website](https://github.com/dlang/dlang.org/).
+DAutoTest tests that the PR doesn't break the documentation generator, and also tests [changelog entries](https://github.com/dlang/dmd/tree/master/changelog) for correct DDoc syntax.
```

Comment:
Does it test the doc generator or test the generated docs? I thought the latter.

Reply:
With 'test generated docs', do you mean test code examples in documentation? In any case, @CyberShadow can you describe what DAutoTest does exactly?

Reply:
- Builds the documentation by invoking the dlang.org makefile.
  - This necessarily also builds the compiler, runtime, standard library, and fetches some dependencies.
- Saves the resulting HTML files and makes them available for preview.
- Allows looking at the differences in generated HTML between the base branch and the PR branch.
- Not DAutoTest but an adjacent component with the same underlying implementation also builds the master branch when it's updated (for any core repository), publishes the result on http://dlang.k3.1azy.net/, and uploads it to the official website.

Reply:
Thanks